stake-pool/program/src/state.rs

```diff
@@ -2,21 +2,38 @@
 
 use {
     crate::{error::StakePoolError, instruction::Fee, processor::Processor},
+    borsh::{BorshDeserialize, BorshSchema, BorshSerialize},
     solana_program::{
-        account_info::AccountInfo, entrypoint::ProgramResult, program_error::ProgramError,
+        account_info::AccountInfo, borsh::get_packed_len, program_error::ProgramError,
         pubkey::Pubkey,
     },
     spl_math::checked_ceil_div::CheckedCeilDiv,
-    std::convert::{TryFrom, TryInto},
-    std::mem::size_of,
+    std::convert::TryFrom,
 };
 
+/// Enum representing the account type managed by the program
+#[derive(Clone, Debug, PartialEq, BorshDeserialize, BorshSerialize, BorshSchema)]
+pub enum AccountType {
+    /// If the account has not been initialized, the enum will be 0
+    Uninitialized,
+    /// Stake pool v1
+    StakePoolV1,
+    /// Validator stake list
+    ValidatorStakeListV1,
```

Comment:
nit: v2 doesn't exist yet and might never exist, no need to allude to it
```suggestion
    /// Stake pool
    StakePool,
    /// Validator stake list
    ValidatorStakeList,
```